src/components/Main.js

```diff
@@ -0,0 +1,25 @@
+import AddContacts from "./AddContacts";
+import IndividualContact from "./IndividualContact";
+import FullList from "./FullList";
+import { Routes, Route, } from 'react-router-dom';
+import React, { useState } from 'react';
+
+const Main = () => {
+  const  [contacts, setContacts] = useState([])
+
+  const mainState = (e) => {
+    setContacts(contacts.concat([e]))
+  }
+
+  return (
+    <div>
+      <Routes>
+        <Route exact path ="/" element={<FullList contacts={contacts}/>}/>
+        <Route path="/add-contacts" element={<AddContacts mainState={mainState}/>}/>
+        <Route path="/individual-contact" element={<IndividualContact />}/>
```

Comment:
You can pass props to the components at each route. That's how I'd get access to an individual contact, for example.